src/csharp/Intel/Generator/Extended/CSharp/CSharpExtendedSyntaxGenerator.cs

```diff
@@ -0,0 +1,701 @@
+					using (writer.Indent()) {
+						foreach (var register in registers.Values) {
+							if (register.Value == 0) {
+								continue;
+							}
+							var name = register.Name(Converter);
+							writer.WriteLine($"public static readonly ExtendedRegister {name.ToLowerInvariant()} = Register.{name};");
+						}
+					}
+					writer.WriteLine("}");
+				}
+				writer.WriteLine("}");
+				writer.WriteLine("#endif");
+			}
+		}
+
+		protected override void Generate(Dictionary<GroupKey, OpCodeInfoGroup> map, OpCodeInfoGroup[] groups) {
+
+			var filename = Path.Combine(CSharpConstants.GetDirectory(_generatorOptions, CSharpConstants.IcedNamespace), "Extended", "ExtendedEncoder.g.cs");
+			using (var writer = new FileWriter(TargetLanguage.CSharp, FileUtils.OpenWrite(filename))) {
+				writer.WriteFileHeader();
+				writer.WriteLine($"#if {CSharpConstants.EncoderDefine}");
+				writer.WriteLine($"namespace {CSharpConstants.IcedNamespace} {{");
+				using (writer.Indent()) {
+					writer.WriteLine("using System;");
+					writer.WriteLine("public sealed partial class ExtendedEncoder {");
+					using (writer.Indent()) {
+
+						var renderArgs = new List<RenderArg>();
+
+						foreach (var group in groups) {
+							renderArgs.Clear();
+
+							int regArgIndex = -1;
+							int rmArgIndex = -1;
+
+							var signature = group.Signature;
+							for (int i = 0; i < signature.ArgCount; i++) {
+								string argName = i == 0 ? "dst" : i == 1 ? "src" : $"arg{i}";
+								string argType = $"arg{i}";
+								var argKind = signature.GetArgKind(i);
+								
+								switch (argKind) {
+								case ArgKind.Register:
+									argType = "Register";
+									if (regArgIndex < 0) {
+										regArgIndex = i;
+									}
+									break;
+								
+								case ArgKind.Branch:
+									argType = "Label";
+									break;
+
+								case ArgKind.RegisterMemory:
+									if (regArgIndex < 0) {
+										regArgIndex = i;
+									}
+
+									if (rmArgIndex < 0) {
+										rmArgIndex = i;
+									}
+									argType = "ExtendedMemoryOperand";
+									break;
+								case ArgKind.Memory:
+									if (rmArgIndex < 0) {
+										rmArgIndex = i;
+									}
+									argType = "ExtendedMemoryOperand";
+									break;
+								case ArgKind.Immediate:
+									argName = $"imm";
+									argType = group.MaxImmediateSize < 8 ? "int" : "long";
+									break;
+								case ArgKind.HiddenMemory:
+									argName = "<none>";
+									argType = "<none>";
+									break;
+								default:
+									throw new ArgumentOutOfRangeException();
+								}
+
+								renderArgs.Add(new RenderArg(argName, argType, argKind));
+							}
+
+							var methodName = Converter.Method(group.Name);
+
+							// Write documentation
+							var methodDoc = new StringBuilder();
+							methodDoc.Append($"{group.Name} instruction.#(p:)#");
+							foreach (var code in group.Items) {
+								if (!string.IsNullOrEmpty(code.Code.Documentation)) {
```

Comment:
Need paragraphs between instruction docs:

```diff
+							bool needParagraph = false;
							foreach (var code in group.Items) {
+								if (needParagraph)
+									methodDoc.Append("#(p:)##(p:)#");
								if (!string.IsNullOrEmpty(code.Code.Documentation)) {
									methodDoc.Append(code.Code.Documentation);
+									needParagraph = true;
								}
							}
```

Reply:
It was not clear to me so a `#(p:)#` is either an openning or closing paragraph, and you need to embrace doc with `#(p:)#this is a paragraph#(p:)#` (because I saw generated `<br/>` which are not really paragraph but line breaks)

Reply:
so `#(p:)#` translates to a single `<br/>` correct?

Reply:
It generates two, one to create a new line and another one to create an empty line, see `CSharpDocCommentWriter` line 114.